students/Harry_Maher/Session05/mailroom3.py

```diff
@@ -0,0 +1,102 @@
+
+    The script should prompt the user (you) to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”)
+
+
+if the user (you) selects ‘Send a Thank You’, prompt for a Full Name.
+
+    If the user types ‘list’, show them a list of the donor names and re-prompt
+    If the user types a name not in the list, add that name to the data structure and use it.
+    If the user types a name in the list, use it.
+    Once a name has been selected, prompt for a donation amount.
+    Turn the amount into a number – it is OK at this point for the program to crash if someone types a bogus amount.
+    Once an amount has been given, add that amount to the donation history of the selected user.
+    Finally, use string formatting to compose an email thanking the donor for their generous donation. Print the email to the terminal and return to the original prompt.
+
+
+    If the user (you) selected “Create a Report”, print a list of your donors, sorted by total historical donation amount.
+        Include Donor Name, total donated, number of donations and average donation amount as values in each row. You do not need to print out all their donations, just the summary info.
+        Using string formatting, format the output rows as nicely as possible. The end result should be tabular (values in each column should align with those above and below)
+        After printing this report, return to the original prompt.
+    At any point, the user should be able to quit their current task and return to the original prompt.
+    From the original prompt, the user should be able to quit the script cleanly.
+
+Your report should look something like this:
+
+Donor Name                | Total Given | Num Gifts | Average Gift
+------------------------------------------------------------------
+William Gates, III         $  653784.49           2  $   326892.24
+Mark Zuckerberg            $   16396.10           3  $     5465.37
+Jeff Bezos                 $     877.33           1  $      877.33
+Paul Allen                 $     708.42           3  $      236.14
+
+
+
+'''
+
+import os
+
+# Just gonna make these global. Lazy and waiting until we get to classes...
+donor_dict = {"Bill":[2.75,3], "Bob": [52.75,15], "Jim": [27.36,2] , "Ann": [12.76,1], "Beth": [31245.75,1]}
+donors = [donor for donor in donor_dict.keys()] 
+
+            
+def send_ty():
+    """Prints out a thank you letter to specified person, or to new person. Doesn't save new person to globally defined dict. We'll wait for oop to do that?"""
+    name = input("Pick or add a name!\n")
+    name = name.title()
+    if name == "list":
+        print("\n".join(donors))
+        name = input("Pick or add a name!")
+    if name not in donors and (name !="list"):
+        donor_dict[name] = [0,0]
+        donation = input("How much did they donate?\n")
+        donor_dict[name][0]+=float(donation)
+        donor_dict[name][1]+=1
+    print(thank_you_txt(name))
```

Comment:
Are you keeping a running total and number of donations? Probably better to just keep all donations, charities are prone to auditing! ;-) But seriously, it generally makes sense to keep raw data. When it gets big enough to be a problem, time to move to a database. :)